ddl/ddl_api.go

```diff
@@ -2779,6 +2851,37 @@ func checkColumnWithIndexConstraint(tbInfo *model.TableInfo, originalCol, newCol
 	return nil
 }
 
+func checkAutoRandom(tableInfo *model.TableInfo, originCol *table.Column, specNewColumn *ast.ColumnDef) error {
+	if !config.AutoRandom.IsAllow() && containsColumnOption(specNewColumn, ast.ColumnOptionAutoRandom) {
```

Comment:
should it be `A || B` here?

Reply:
Using `&&` is correct: only when the config says it's not allowed to use `auto_random` __and__ the user specifies `auto_random`, report an error.

Reply:
ok, so if config is allowed to use `auto_random` and the user specifies no `auto_random`, it still go down the check.